[dv] Update doc for TLUL integrity (#7686)
lowRISC/opentitan: hw/dv/sv/cip_lib/doc/index.md

* **ral**: This is the instance to the auto-generated RAL model that is
  extended from `dv_base_reg_block`. In the base class, this is created using
  the RAL_T class parameter which the extended IP env cfg class sets correctly.
* **tl_intg_alert_name**: This the name of the alert that will be triggered once
  a TLUL integrity error occurs. If the block isn't used the default alert name

>   `fatal_fault`, the user should override it at `initialize`

Reviewer: maybe follow the previous bullet point and add a period? same for line 77

Author: Good point. Updated, thanks

> * **tl_intg_alert_name**: This the name of the alert that will be triggered once
>   a TLUL integrity error occurs. If the block isn't used the default alert name
>   `fatal_fault`, the user should override it at `initialize`.

Reviewer 2: ```suggestion
* **tl_intg_alert_name**: Name of the alert that will be triggered on
  TLUL integrity error detection. The default name used for this type
  of alert is "fatal_fault". The block may use a different name too - in
  that case, please update this member to reflect the correct name 
  in the `initialize()` method.  
```

> * **tl_intg_alert_fields**: When a TLUL integrity error triggers some of CSR
>   status changed, the user should update this associative array at `initialize` as
>   follows.
> ```systemverilog
> tl_intg_alert_fields[reg_name] = field_name;

Reviewer 2: ```suggestion
* **tl_intg_alert_fields**: An associative array of CSR fields keyed with the 
  corresponding register name. This is the list of CSR fields that are modified 
  when an alert triggers due to TL integrity violation event. The DV user is 
  required to build this list in the `initialize()` method. 

```systemverilog
tl_intg_alert_fields[reg_name] = ral.foo_csr.bar_field;
```

Question: By `reg_name` do you mean the alert status CSR name itself, or is this the reg block name (unique to each TL device interface)?

Author: I meant the csr name itself, but I changed my mind to use the object handle as key now, so that we can check if the field path is correct in compile and supply the value as well. Please take a look at #7677.

I also updated the wording as you suggested, PTAL. Thanks

```

> * **en_tl_intg_gen**: This is a knob to enable/disable TLUL integrity
>   genearation in sequences and corresponding checkers in scoreboard.

Reviewer 2: ```suggestion
* **en_tl_intg_gen**: Controls whether the TLUL integrity error is generated in
   sequences / checked by the scoreboard.
```

* **ral**: This is the instance to the auto-generated RAL model that is
  extended from `dv_base_reg_block`. In the base class, this is created using
  the RAL_T class parameter which the extended IP env cfg class sets correctly.
* **tl_intg_alert_name**: Name of the alert that will be triggered on  TLUL
  integrity error detection. The default name used for this type of alert is

>   "fatal_fault". The block may use a different name too - in that case, please
>   update this member to reflect the correct name in the `initialize()` method.

Reviewer 3: ```suggestion
  "fatal_fault". Update this variable in the 'initialize() method to change the block's alert name.
```

Author: prefer to clearly say to update it when the name is different from default.

  CSR/memory value.
* **task run_tl_intg_err_vseq**: This task will test TLUL integrity error. It contains

>   2 parallel threads. One thread calls `csr_rw` seq to do random CSR accesses. The

Reviewer 2: ```suggestion
  2 parallel threads. The first one invokes the `csr_rw` seq to drive random, legal CSR accesses. The
```

>   other issues a TLUL item with flipping up to 3 bits on the integrity fields (
>   `a_user.cmd_intg`, `a_user.data_intg`) or other TLUL a_chann fields, and then check
>   that DUT returns `d_error = 1` and the fatal alert is triggered.
>   If some CSRs are affected by a TLUL integrity error, the user should provide the
>   register names and the field names at cfg.tl_intg_alert_fields.

Reviewer 2: ```suggestion
  second drives a bad TLUL transaction that violates the payload integrity. The bad 
  packet is created by corrupting upto 3 bits either in the integrity (ECC) fields 
  (`a_user.cmd_intg`, `a_user.d_intg`), or in their corresponding command / data 
  payload itself. The sequence then verifies that the DUT not only returns an error 
  response (with `d_error` = 1), but also triggers a fatal alert. The list of CSRs that 
  are impacted by this alert event maintained in `cfg.tl_intg_alert_fields`  are also 
  checked for correctness.
 ```

  etc. All the items sent in this task will trigger d_error and won't change the
  CSR/memory value.
* **task run_tl_intg_err_vseq**: This task will test TLUL integrity error. It contains
  2 parallel threads. The first one invokes the `csr_rw` seq to drive random, legal
  CSR accesses. The second drives a bad TLUL transaction that violates the payload
  integrity. The bad packet is created by corrupting upto 3 bits either in the integrity
  (ECC) fields (`a_user.cmd_intg`, `a_user.d_intg`), or in their corresponding command /
  data payload itself. The sequence then verifies that the DUT not only returns an error
  response (with `d_error` = 1), but also triggers a fatal alert. The list of CSRs that

>   are impacted by this alert event maintained in `cfg.tl_intg_alert_fields` are also

Reviewer 3: ```suggestion
  are impacted by this alert event, maintained in `cfg.tl_intg_alert_fields`, are also
```

Author: Updated, thanks
